src/ShaderTools/Hlsl/Syntax/StructTypeSyntax.cs

```diff
@@ -13,8 +13,10 @@ public class StructTypeSyntax : TypeDefinitionSyntax
 
         public override SyntaxToken NameToken => Name;
 
+        public bool IsClass => Kind == SyntaxKind.ClassKeyword;
+
         public StructTypeSyntax(SyntaxToken structKeyword, SyntaxToken name, BaseListSyntax baseList, SyntaxToken openBraceToken, List<SyntaxNode> members, SyntaxToken closeBraceToken)
-            : base(SyntaxKind.StructType)
+            : base(structKeyword.Kind)
```

Comment:
This should stay as `StructType`, because it's describing the overall `StructTypeSyntax` object, not the keyword kind.